examples/common/utility/utility.hpp

```diff
@@ -356,6 +358,46 @@ class cli_argument_pack {
     }
 }; // class cli_argument_pack
 
+// utility class to aid relative error measurement of samples
+class measurements {
+public:
+    measurements() {
+        clear();
+    }
+    void clear() {
+        _secPerFrame.clear();
+    }
+    void start() {
+        _startTime = std::chrono::steady_clock::now();
+    }
+    void stop() {
+        _endTime = std::chrono::steady_clock::now();
+        std::chrono::duration<double> duration = _endTime - _startTime;
+        // store the duration in seconds
+        _secPerFrame.push_back(duration.count());
+    }
+    double computeRelError() {
+        auto averageTimePerFrame =
+            std::accumulate(_secPerFrame.begin(), _secPerFrame.end(), 0.0) / _secPerFrame.size();
+        std::vector<double> diff(_secPerFrame.size());
+        std::transform(_secPerFrame.begin(),
+                       _secPerFrame.end(),
+                       diff.begin(),
+                       [averageTimePerFrame](double x) {
+                           return (x - averageTimePerFrame);
+                       });
+        double sumOfSquareDiff = std::inner_product(diff.begin(), diff.end(), diff.begin(), 0.0);
```

Comment:
We can also transform the data in place. This will save us one pass through the whole vector.
```suggestion
        double sumOfSquareDiff = 0.0;
        std::for_each(_secPerFrame.begin(),
                      _secPerFrame.end(),
                       [averageTimePerFrame, &sumOfSquareDiff](double x) {
                           double diff = (x - averageTimePerFrame);
                           sumOfSquareDiff += diff * diff;
                       });
```